configure.ac

```diff
@@ -1078,7 +1078,55 @@ if test "$nettle_cv_asm_x86_gnu_property" = yes; then
 	.popsection'
 fi
 
+dnl  Define
+dnl Determine if aarch64 PAC and/or BTI support should be enabled.
+dnl  Define ASM_AARCH64_PACBTI_ENABLE:
+dnl     - Bit 0: BTI on off
+dnl     - Bit 1: PAC A Key Enabled
+dnl     - Bit 2: PAC B key Enabled
+dnl  Note: Bit 1 and 2 are mutually exclusive.
+dnl  See: https://developer.arm.com/documentation/101028/0012/5--Feature-test-macros
+ASM_AARCH64_PACBTI_ENABLE=0
+AC_COMPILE_IFELSE([
+  AC_LANG_PROGRAM([
+    #if !defined(__ARM_FEATURE_BTI_DEFAULT) && __ARM_FEATURE_BTI_DEFAULT != 1
+    #error "BTI is not enabled"
+    #endif
+  ])
+],
+[ASM_AARCH64_PACBTI_ENABLE=$(( $ASM_AARCH64_PACBTI_ENABLE | 0x1 ))],
```

Comment:
Arithmetic expressions in shell script are still not [recommended](https://www.gnu.org/savannah-checkouts/gnu/autoconf/manual/autoconf-2.71/html_node/Shell-Substitutions.html#index-_0024_0028_0028expression_0029_0029) for the use with Autoconf. Given the variable is only consumed at m4 level, maybe you could simply create a separate variable for each, e.g., `ASM_AARCH64_PACBTI_PAC_A_KEY`, ...)?

Reply:
I think what it's really saying is to use `AS_VAR_ARITH` and base 10 constants.

Reply:
@ueno the more I thought about it, the more I liked your suggestion. Considering that PAC is only used to set the gnu notes section, It greatly reduced the patch size and complexity. Changes forthcoming at some point.